src/lib/api/org/index.ts

```diff
@@ -275,3 +277,108 @@ function convertToSnykProject(projectData: v3ProjectData[]) : SnykProject[] {
   return projects
 }
 
+export async function listTargets(
+  requestManager: requestsManager,
+  orgId: string,
+  config?: {
+    limit?: number,
+    isEmpty?: boolean,
+    origin?: string,
+  }
+): Promise<TargetsResponse> {
+  getApiToken();
+  getSnykHost();
+  debug(`Listing all targets for org: ${orgId}`);
+
+  if (!orgId) {
+    throw new Error(
+      `Missing required parameters. Please ensure you have set: orgId, settings.
+        \nFor more information see: https://apidocs.snyk.io/?version=2022-09-15%7Ebeta#get-/orgs/-org_id-/targets`,
+    );
+  }
+  
+  const targets = await listAllSnykTarget(requestManager, orgId, config)
+
+  return {targets: targets};
+}
+
+export async function listAllSnykTarget(requestManager: requestsManager,
+  orgId: string,
+  config?: {
+    limit?: number,
+    isEmpty?: boolean,
+    origin?: string,
+  }
+  ): Promise<SnykTarget[]> {
+
+    let lastPage = false;
+    let SnykTargetList: SnykTarget[] = [];
+    let nextPageLink: string | undefined = undefined;
+    while (!lastPage) {
+      try {
+        const {
+          SnykTarget,
+          next,
+        }: {
+          SnykTarget: SnykTarget[];
+          next?: string;
+        } = await getSnykTarget(requestManager, orgId, nextPageLink, config);
+
+        SnykTargetList = SnykTargetList.concat(SnykTarget);
+        next
+          ? ((lastPage = false), (nextPageLink = next))
+          : ((lastPage = true), (nextPageLink = ''));
+      } catch (e) {
+        debug('Failed to get targets for ', orgId, e);
+        throw e;
+    }
+  }
+  return SnykTargetList
+}
+
+export async function getSnykTarget(requestManager: requestsManager,
+  orgId: string,
+  nextPageLink?: string, 
+  config?: {
+    limit?: number,
+    isEmpty?: boolean,
+    origin?: string,
+  }): Promise< { SnykTarget: SnykTarget[], next?: string } > {
+
+    const link = nextPageLink ? nextPageLink : `https://api.dev.snyk.io/rest/orgs/${orgId.trim()}/targets?version=2022-09-15~beta`
+    const url = new URL(link)
+
+    if (config) {
+      if (config.isEmpty) {
+        url.searchParams.set('isEmpty', String(config.isEmpty))
+      } 
+      if (config.limit) {
+        url.searchParams.set('limit', String(config.limit))
+      } 
+      if (config.origin) {
+        url.searchParams.set('origin', config.origin)
+      }       
+    }
+    
+    const res = await requestManager.request({
```

Comment:
We're mostly using the REST endpoints, shouldn't the default for "useRESTApi" be true and will save us sending this?

Reply:
It is not true in this tool we are, in others we don't. also for the updateProject function, we will use v1 I think we are still using v1 more. also this would be a change in the requests manager not in this tool